internal/fsutil/create_test.go

```diff
@@ -184,17 +194,23 @@ func (s *S) TestCreate(c *C) {
 		c.Assert(err, IsNil)
 		c.Assert(testutil.TreeDump(dir), DeepEquals, test.result)
 
-		if entry.Link != "" && entry.Mode&fs.ModeSymlink == 0 {
-			// Entry is a hardlink.
+		isHardLink := entry.Link != "" && entry.Mode&fs.ModeSymlink == 0
+		// Special case for hard links: use the same file info to compare the
+		// path and the link.
+		if isHardLink {
 			pathInfo, err := os.Lstat(entry.Path)
 			c.Assert(err, IsNil)
 			linkInfo, err := os.Lstat(entry.Link)
 			c.Assert(err, IsNil)
 			os.SameFile(pathInfo, linkInfo)
+		}
+		// [fsutil.Create] does not return information about parent directories
+		// created implicitly. We only check for the requested path.
+		entry.Path = strings.TrimPrefix(entry.Path, dir)
+		if isHardLink {
+			result := fmt.Sprintf("hard link %#o %s", entry.Mode.Perm(), entry.Link)
+			c.Assert(testutil.TreeDumpEntry(entry), DeepEquals, result)
```

Comment:
Let's think about what we are trying to do there. I see that you want to add the check that the TreeDumpEntry has the hardlink information. However, think about where the information is coming from here. We are checking `testutil.TreeDumpEntry()` against `result`.

The first one is a test function and not any real code and the result is something that we just fabricated, so we are not testing any real code. As for the entry itself, we are already checking that the existing entry is a hardlink in the block above using `os.SameFile`.

Because this is only testing the `testutil` implementation I think it is better to remove it.

Reply:
Removed testing against `TreeDumpEntry`, but kept the changes in it. [4090b4c](https://github.com/letFunny/chisel/pull/17/commits/4090b4c4c27eda1bd386bb1757572a4529887ad9)